app/services/offer_validations.rb

```diff
@@ -50,11 +50,11 @@ def restrict_reverting_rejection
       errors.add(:base, :application_rejected_by_default)
     end
 
-    if !candidate_in_apply_2? && application_choice.application_form.apply_1? && any_accepted_offers?
+    if any_accepted_offers?
       errors.add(:base, :other_offer_already_accepted)
     end
 
-    if candidate_in_apply_2? && !application_choice.candidate.current_application_choice.eql?(application_choice)
+    if candidate_in_apply_2? && application_choice.candidate.current_application_choices.exclude?(application_choice)
```

Comment:
Should this whole validation be skipped if the new “3 goes at Apply again” feature flag is active? I had thought that the intention was for the same rules to be used in apply again as on the initial application?

Reply:
This validation is meant to work on both implementation of Apply 2 (1 and up to 3 applications). 

In manage reverting rejections is not allowed, however in the API it is as some vendors required this feature. To prevent them attempting to revert applications in Apply 1 when a candidate is already in Apply 2, this validation kicks in. 

To support Apply 2 multiple applications change, we instead check for the application choice that's being attempted to be in the current list of applications, rather than a single one, which started erroring now even in manage, when attempting to make an offer on an application in apply 2 with multiple applications. 

This ensures that in both versions of Apply 2 we will raise the correct error, while avoiding raising an error when a genuine offer is made in manage or the API.

Reply:
Do we need the `candidate_in_apply_2?` check? Given that for the targeted application choice not to be in the set of current application choices, the set of _current_ application choices must be on a different application, so there must be at least 2 applications, so they must be in apply 2?

Reply:
addressed

Reply:
Since for now we are still using this logic, I think its ok to keep it. If we move away from phases, happy to revisit this then